app/src/main/java/com/asma/tasky/feature_authentication/data/remote/util/ApiKeyInterceptor.kt

```diff
@@ -0,0 +1,17 @@
+package com.asma.tasky.feature_authentication.data.remote.util
+
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+
+class ApiKeyInterceptor : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val newRequest: Request = chain.request().newBuilder()
+
+            .addHeader("x-api-key", "10895044a3d34ddba7fa77f292d145fe")
```

Comment:
I'd recommend passing in this API key from a file that is not in version control to protect such secrets, especially in public repos. Personally, I like to read it in as a build config field from `local.properties` and then read it in the app like this: `BuildConfig.API_KEY`